### PR TITLE
feat (bfdr): Facility/Location

### PR DESCRIPTION
**Pull request by RobertScalfani**
This PR implements:

- FNPI (FacilityNPI)
- SFN (FacilityJFI)
- HOSP (BirthFacilityName)
- HOSPFROM (FacilityMotherTransferredFrom)
- HOSPTO (FacilityInfantTransferredTo)

It also updates the URLs to include the `/Location-bfdr` URL. All URLs were updated according to the stu2-ballot branch of the BFDR IG.

### CODE REVIEW

**Comment by smacadam** [comment]
when you generated the URLs did you use Pete's new script? the new script has been merged into Natality-feature

**Comment by RobertScalfani** [comment]
> when you generated the URLs did you use Pete's new script? the new script has been merged into Natality-feature

Yup, URLs are all set

File: projects/BFDR/BirthRecord_submissionProperties.cs [hunk 1]
```diff
@@ -4907,6 +4907,160 @@ public string FatherEducationLevelEditFlagHelper
...
+
+            if (entry != null)
+            {
+                Observation obs = entry.Resource as Observation;
+                return obs;
+            }
+            return null;
+        }
+
+        private Observation CreateObservationEntry(string loincCode, string subjectId, string compositionSection, string focusId = null)
```
Comment (smacadam) [inline projects/BFDR/BirthRecord_submissionProperties.cs:5047]: This helper was refactored into VitalRecord_util in a previous merged PR. There should be a function in there called GetOrCreateObservation that replaced this one. This one can be removed.

File: projects/BFDR/BirthRecord_submissionProperties.cs [hunk 2]
```diff
@@ -4907,6 +4907,160 @@ public string FatherEducationLevelEditFlagHelper
...
+        {
+            get => GetFacilityLocation(ValueSets.LocationTypes.Transfer_To_Location)?.Name;
+            set => (GetFacilityLocation(ValueSets.LocationTypes.Transfer_To_Location) ?? CreateAndSetLocationBirth(ValueSets.LocationTypes.Transfer_To_Location)).Name = value;
+        }
+
+        private Location GetFacilityLocation(string code) {
+            return (Location)Bundle.Entry.Where(e => e.Resource is Location loc && loc.Type.Any(type => type.Coding.Any(coding => coding.System == CodeSystems.LocalBFDRCodes && coding.Code == code))).FirstOrDefault()?.Resource;
+        }
+        
+        private Observation GetObservation(string code)
```
Comment (smacadam) [inline projects/BFDR/BirthRecord_submissionProperties.cs:5032]: This helper was refactored into VitalRecord_util in a previous merged PR. There should be a function in there called GetObservation that replaced this one. This one can be removed.

File: projects/BFDR/BirthRecord_submissionProperties.cs (hunk 1 above)
Comment (RobertScalfani) [reply projects/BFDR/BirthRecord_submissionProperties.cs]: Whoops, must've missed that during a merge. Resolved.

File: projects/BFDR/BirthRecord_submissionProperties.cs (hunk 2 above)
Comment (RobertScalfani) [reply projects/BFDR/BirthRecord_submissionProperties.cs]: Resolved.

File: projects/BFDR/BirthRecord_constructors.cs [hunk 3]
```diff
@@ -160,7 +160,7 @@ public Bundle GetDemographicCodedContentBundle()
             dccBundle.Type = Bundle.BundleType.Collection;
             dccBundle.Meta = new Meta();
             // TODO: URLs.cs has the profile URL with the extensions instead of the profiles, fix this once URLs.cs is fixed
```
Comment (smacadam) [inline projects/BFDR/BirthRecord_constructors.cs:162]: should this TODO be removed?

**Comment by RobertScalfani** [comment]
> Looks good! just one todo that looks like it can be removed

Should be good to go now

**Review by smacadam — APPROVED** [review]

**Pull request merged**